iceoryx_utils/test/moduletests/test_cxx_vector.cpp

```diff
@@ -1036,3 +1043,42 @@ TEST_F(vector_test, PartiallyEqualVectorsWithDifferentCapacityAreNotEqual)
     EXPECT_FALSE(a == b);
     EXPECT_TRUE(a != b);
 }
+
+TEST_F(vector_test, FullVectorDestroysElementsInReverseOrder)
+{
+    static constexpr uint64_t VECTOR_CAPACITY = 35U;
+    {
+        vector<CTorTest, VECTOR_CAPACITY> sut;
+
+        for (uint64_t i = 0U; i < VECTOR_CAPACITY; ++i)
+        {
+            sut.emplace_back(i);
+        }
+    }
+
+    ASSERT_THAT(dtorOrder.size(), Eq(VECTOR_CAPACITY));
+    for (uint64_t i = 0U; i < VECTOR_CAPACITY; ++i)
+    {
+        EXPECT_THAT(dtorOrder[i], Eq(VECTOR_CAPACITY - i - 1U));
```

Comment:
a `static constexpr uint64_t INDEX_END = VECTOR_CAPACITY - 1U;` or similar would make this clearer